src/Form/input-classifiers.js

```diff
@@ -10,18 +10,19 @@ const {ATTR_INPUT_TYPE} = require('../constants')
 /**
  * Tries to get labels even when they're not explicitly set with for="id"
  * @param el
+ * @param {Form} form
  * @returns {[]|NodeList}
  */
-const findLabels = (el) => {
+const findLabels = (el, form) => {
     const parentNode = el.parentNode
-    if (!parentNode) return []
+    if (!parentNode || parentNode === form.form || el === form.form) return []
```

Comment:
In this file, I've started passing the form class to these calls, so we can restrict the search for labels to the parent form (or the equivalent element we identified). This prevents the search bubbling up potentially to the whole body, thus causing false positives.

It looks rather noisy, with a lot of parameter passing. In the future, we might introduce a wrapper class around the input that would contain all of this info, but it's out of scope for this change. Also, deriving the form every time can be expensive if a standard form is not present (we traverse the DOM upwards to find some kind of container).

Reply:
What happens for inputs without a form? Will this be undefined or do we not handle those kinda forms?

Reply:
We try to identify a container by traversing the DOM upwards and searching for buttons and other input fields. If we don't have a plausible form, the use the input itself as the "form" element.

Reply:
I think that's probably fine as `parentNode === form.form || el === form.form` will essentially be: `parentNode === null || el === null` and we'll just treat the whole DOM as a the form which appears to be the right thing.